PyPortal_Azure_Plant_Monitor/code.py

```diff
@@ -31,18 +38,36 @@
 esp32_reset = DigitalInOut(board.ESP_RESET)
 spi = busio.SPI(board.SCK, board.MOSI, board.MISO)
 esp = adafruit_esp32spi.ESP_SPIcontrol(spi, esp32_cs, esp32_ready, esp32_reset)
+
+# Set up the WiFi manager with a status light to show the WiFi connection status
 status_light = neopixel.NeoPixel(board.NEOPIXEL, 1, brightness=0.2)
 wifi = adafruit_esp32spi_wifimanager.ESPSPI_WiFiManager(esp, secrets, status_light)
+wifi.connect()
+
+# Time setup, needed to authenticate with Azure IoT Central
+ntp = NTP(esp)
+while not ntp.valid_time:
```

Comment:
may want to print `Could not get valid time, retrying in 5s...`

Reply:
something like: https://github.com/adafruit/Adafruit_CircuitPython_NTP/blob/master/examples/ntp_simpletest.py#L34